### PR TITLE
Copy HTTP load tests from ballerina-performance cloud to the HTTP repository

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2520](https://github.com/ballerina-platform/ballerina-standard-library/issues/2520)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: load-tests/h1_h1_passthrough/deployment/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h1-h1-passthrou-deployment
+- path: secret-patch.yaml
+  target:
+    version: v1
+    kind: Secret
```
Comment (dilanSachi) [inline load-tests/h1_h1_passthrough/deployment/kustomization.yaml:15]: newline

File: load-tests/h1_h1_passthrough/deployment/netty-backend.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h1_h1_passthrough/deployment/netty-backend.yaml:35]: newline

File: load-tests/h1_h1_passthrough/scripts/http-post-request.jmx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            <stringProp name="Assertion.test_field">Assertion.response_data</stringProp>
+            <boolProp name="Assertion.assume_success">false</boolProp>
+            <intProp name="Assertion.test_type">16</intProp>
+          </ResponseAssertion>
+          <hashTree/>
+        </hashTree>
+      </hashTree>
+    </hashTree>
+  </hashTree>
+</jmeterTestPlan>
```
Comment (dilanSachi) [inline load-tests/h1_h1_passthrough/scripts/http-post-request.jmx:80]: newline

File: load-tests/h1_transformation/deployment/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h1-transformati-deployment
+- path: secret-patch.yaml
+  target:
+    version: v1
+    kind: Secret
```
Comment (dilanSachi) [inline load-tests/h1_transformation/deployment/kustomization.yaml:15]: newline

File: load-tests/h1_transformation/deployment/netty-backend.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h1_transformation/deployment/netty-backend.yaml:35]: newline

File: load-tests/h1_transformation/scripts/http-post-request.jmx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            <stringProp name="Assertion.test_field">Assertion.response_code</stringProp>
+            <boolProp name="Assertion.assume_success">false</boolProp>
+            <intProp name="Assertion.test_type">16</intProp>
+          </ResponseAssertion>
+          <hashTree/>
+        </hashTree>
+      </hashTree>
+    </hashTree>
+  </hashTree>
+</jmeterTestPlan>
```
Comment (dilanSachi) [inline load-tests/h1_transformation/scripts/http-post-request.jmx:80]: newline

File: load-tests/h1c_h1c_passthrough/deployment/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  - h1c_h1c_passthrough.yaml
+  - ingress.yaml
+  - netty-backend.yaml
+patches:
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h1c-h1c-passthr-deployment
```
Comment (dilanSachi) [inline load-tests/h1c_h1c_passthrough/deployment/kustomization.yaml:11]: newline

File: load-tests/h1c_h1c_passthrough/deployment/netty-backend.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h1c_h1c_passthrough/deployment/netty-backend.yaml:32]: newline

File: load-tests/h1c_h1c_passthrough/scripts/http-post-request.jmx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            <stringProp name="Assertion.test_field">Assertion.response_data</stringProp>
+            <boolProp name="Assertion.assume_success">false</boolProp>
+            <intProp name="Assertion.test_type">16</intProp>
+          </ResponseAssertion>
+          <hashTree/>
+        </hashTree>
+      </hashTree>
+    </hashTree>
+  </hashTree>
+</jmeterTestPlan>
```
Comment (dilanSachi) [inline load-tests/h1c_h1c_passthrough/scripts/http-post-request.jmx:80]: check other places also

File: load-tests/h1_h1_passthrough/scripts/run.sh [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ----------------------------------------------------------------------------
+# Execution script for ballerina performance tests
+# ----------------------------------------------------------------------------
+set -e
+source base-scenario.sh
+
+jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=1200 -Jhost=bal.perf.test -Jport=443 -Jprotocol=https -Jpath=passthrough $payload_flags
```
Comment (dilanSachi) [inline load-tests/h1_h1_passthrough/scripts/run.sh:22]:
```suggestion
jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=3600 -Jhost=bal.perf.test -Jport=443 -Jprotocol=https -Jpath=passthrough $payload_flags
```

File: load-tests/h1c_h1c_passthrough/scripts/http-post-request.jmx (hunk 9 above)
Comment (ayeshLK) [reply load-tests/h1c_h1c_passthrough/scripts/http-post-request.jmx]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1c_h1c_passthrough/deployment/netty-backend.yaml (hunk 8 above)
Comment (ayeshLK) [reply load-tests/h1c_h1c_passthrough/deployment/netty-backend.yaml]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1c_h1c_passthrough/deployment/kustomization.yaml (hunk 7 above)
Comment (ayeshLK) [reply load-tests/h1c_h1c_passthrough/deployment/kustomization.yaml]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_transformation/scripts/http-post-request.jmx (hunk 6 above)
Comment (ayeshLK) [reply load-tests/h1_transformation/scripts/http-post-request.jmx]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_transformation/deployment/netty-backend.yaml (hunk 5 above)
Comment (ayeshLK) [reply load-tests/h1_transformation/deployment/netty-backend.yaml]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_transformation/deployment/kustomization.yaml (hunk 4 above)
Comment (ayeshLK) [reply load-tests/h1_transformation/deployment/kustomization.yaml]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_h1_passthrough/scripts/http-post-request.jmx (hunk 3 above)
Comment (ayeshLK) [reply load-tests/h1_h1_passthrough/scripts/http-post-request.jmx]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_h1_passthrough/deployment/netty-backend.yaml (hunk 2 above)
Comment (ayeshLK) [reply load-tests/h1_h1_passthrough/deployment/netty-backend.yaml]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_h1_passthrough/deployment/kustomization.yaml (hunk 1 above)
Comment (ayeshLK) [reply load-tests/h1_h1_passthrough/deployment/kustomization.yaml]: Fixed with [a0c12388f9a41e63587bb2d29ed514cb9a005fd9](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/a0c12388f9a41e63587bb2d29ed514cb9a005fd9)

File: load-tests/h1_transformation/scripts/run.sh [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ----------------------------------------------------------------------------
+# Execusion script for ballerina performance tests
+# ----------------------------------------------------------------------------
+set -e
+source base-scenario.sh
+
+jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=1200 -Jhost=bal.perf.test -Jport=443 -Jprotocol=https -Jpath=transform $payload_flags
```
Comment (ayeshLK) [inline load-tests/h1_transformation/scripts/run.sh:22]:
```suggestion
jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=3600 -Jhost=bal.perf.test -Jport=443 -Jprotocol=https -Jpath=transform $payload_flags
```

File: load-tests/h1c_h1c_passthrough/scripts/run.sh [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ----------------------------------------------------------------------------
+# Execution script for ballerina performance tests
+# ----------------------------------------------------------------------------
+set -e
+source base-scenario.sh
+
+jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=1200 -Jhost=bal.perf.test -Jport=80 -Jprotocol=http -Jpath=passthrough $payload_flags
```
Comment (ayeshLK) [inline load-tests/h1c_h1c_passthrough/scripts/run.sh:22]:
```suggestion
jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=3600 -Jhost=bal.perf.test -Jport=80 -Jprotocol=http -Jpath=passthrough $payload_flags
```

File: load-tests/h1c_transformation/scripts/run.sh [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ----------------------------------------------------------------------------
+# Execution script for ballerina performance tests
+# ----------------------------------------------------------------------------
+set -e
+source base-scenario.sh
+
+jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=1200 -Jhost=bal.perf.test -Jport=80 -Jprotocol=http -Jpath=transform $payload_flags
```
Comment (ayeshLK) [inline load-tests/h1c_transformation/scripts/run.sh:22]:
```suggestion
jmeter -n -t "$scriptsDir/"http-post-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=3600 -Jhost=bal.perf.test -Jport=80 -Jprotocol=http -Jpath=transform $payload_flags
```

File: load-tests/observability_enabled/scripts/run.sh [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ----------------------------------------------------------------------------
+# Execusion script for ballerina performance tests
+# ----------------------------------------------------------------------------
+set -e
+source base-scenario.sh
+jmeter -n -t "$scriptsDir/"http-get-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=1200 -Jhost=bal.perf.test -Jport=80 -Jprotocol=http -Jpath=hello
```
Comment (ayeshLK) [inline load-tests/observability_enabled/scripts/run.sh:21]:
```suggestion
jmeter -n -t "$scriptsDir/"http-get-request.jmx -l "$resultsDir/"original.jtl -Jusers="$concurrent_users" -Jduration=3600 -Jhost=bal.perf.test -Jport=80 -Jprotocol=http -Jpath=hello
```

File: load-tests/h2_h2_passthrough/deployment/netty-backend.yaml [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h2_h2_passthrough/deployment/netty-backend.yaml:36]: here also

File: load-tests/h2_h2_passthrough/deployment/kustomization.yaml [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h2-h2-passthrou-deployment
+- path: secret-patch.yaml
+  target:
+    version: v1
+    kind: Secret
```
Comment (dilanSachi) [inline load-tests/h2_h2_passthrough/deployment/kustomization.yaml:15]: .

File: load-tests/h2_h1c_passthrough/deployment/netty-backend.yaml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h2_h1c_passthrough/deployment/netty-backend.yaml:34]: .

File: load-tests/h2_h1c_passthrough/deployment/kustomization.yaml [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+  - h2_h1c_passthrough.yaml
+  - ingress.yaml
+  - netty-backend.yaml
+patches:
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h2-h1c-passthro-deployment
```
Comment (dilanSachi) [inline load-tests/h2_h1c_passthrough/deployment/kustomization.yaml:11]: .

File: load-tests/h2_h1_passthrough/deployment/netty-backend.yaml [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h2_h1_passthrough/deployment/netty-backend.yaml:34]: .

File: load-tests/h2_h1_passthrough/deployment/kustomization.yaml [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h2-h1-passthrou-deployment
+- path: secret-patch.yaml
+  target:
+    version: v1
+    kind: Secret
```
Comment (dilanSachi) [inline load-tests/h2_h1_passthrough/deployment/kustomization.yaml:15]: .

File: load-tests/h1c_transformation/scripts/http-post-request.jmx [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+            <stringProp name="Assertion.test_field">Assertion.response_data</stringProp>
+            <boolProp name="Assertion.assume_success">false</boolProp>
+            <intProp name="Assertion.test_type">16</intProp>
+          </ResponseAssertion>
+          <hashTree/>
+        </hashTree>
+      </hashTree>
+    </hashTree>
+  </hashTree>
+</jmeterTestPlan>
```
Comment (dilanSachi) [inline load-tests/h1c_transformation/scripts/http-post-request.jmx:80]: .

File: load-tests/h1c_transformation/deployment/netty-backend.yaml [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: netty
+spec:
+  type: ClusterIP
+  ports:
+  - port: 8688
+  selector:
+    app: netty-backend
```
Comment (dilanSachi) [inline load-tests/h1c_transformation/deployment/netty-backend.yaml:32]:
```suggestion
    app: netty-backend
    
```

File: load-tests/h1c_transformation/deployment/kustomization.yaml [hunk 23]
```diff
@@ -0,0 +1,11 @@
...
+  - h1c_transformation.yaml
+  - ingress.yaml
+  - netty-backend.yaml
+patches:
+- path: deployment-patch.yaml
+  target:
+    group: apps
+    version: v1
+    kind: Deployment
+    name: h1c-transformat-deployment
```
Comment (dilanSachi) [inline load-tests/h1c_transformation/deployment/kustomization.yaml:11]:
```suggestion
    name: h1c-transformat-deployment
    
```

File: load-tests/h2_h2_passthrough/deployment/netty-backend.yaml (hunk 15 above)
Comment (ayeshLK) [reply load-tests/h2_h2_passthrough/deployment/netty-backend.yaml]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

File: load-tests/h2_h1c_passthrough/deployment/netty-backend.yaml (hunk 17 above)
Comment (ayeshLK) [reply load-tests/h2_h1c_passthrough/deployment/netty-backend.yaml]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

File: load-tests/h2_h1_passthrough/deployment/netty-backend.yaml (hunk 19 above)
Comment (ayeshLK) [reply load-tests/h2_h1_passthrough/deployment/netty-backend.yaml]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

File: load-tests/h1c_transformation/scripts/http-post-request.jmx (hunk 21 above)
Comment (ayeshLK) [reply load-tests/h1c_transformation/scripts/http-post-request.jmx]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

File: load-tests/h2_h2_passthrough/deployment/kustomization.yaml (hunk 16 above)
Comment (ayeshLK) [reply load-tests/h2_h2_passthrough/deployment/kustomization.yaml]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

File: load-tests/h2_h1c_passthrough/deployment/kustomization.yaml (hunk 18 above)
Comment (ayeshLK) [reply load-tests/h2_h1c_passthrough/deployment/kustomization.yaml]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

File: load-tests/h2_h1_passthrough/deployment/kustomization.yaml (hunk 20 above)
Comment (ayeshLK) [reply load-tests/h2_h1_passthrough/deployment/kustomization.yaml]: Fixed with [e389f9461f5d91919b80582086e890a9825e7fb7](https://github.com/ballerina-platform/module-ballerina-http/pull/793/commits/e389f9461f5d91919b80582086e890a9825e7fb7)

**Review by dilanSachi — APPROVED** [review]
LGTM

**Pull request merged**